File: backend/app/schemas/summary.py
from pydantic import BaseModel


class CategorySummary(BaseModel):
    name: str
    emoji: str
    total: float
    count: int


class DailySummary(BaseModel):
    date: str
    total: float


class MonthlySummaryResponse(BaseModel):
    year: int
    month: int
    total: float
    count: int
    by_category: list[CategorySummary]
    daily: list[DailySummary]

File: frontend/src/services/summary.ts
import api from "./api"

export interface CategorySummary {
  name: string
  emoji: string
  total: number
  count: number
}

export interface DailySummary {
  date: string
  total: number
}

export interface MonthlySummary {
  year: number
  month: number
  total: number
  income_total: number
  count: number
  by_category: CategorySummary[]
  daily: DailySummary[]
}

export const summaryApi = {
  monthly: (year?: number, month?: number) =>
    api.get<MonthlySummary>("/summary/monthly", {
      params: { ...(year && { year }), ...(month && { month }) },
    }),
}
